app/controllers/cinema_halls_controller.rb

```diff
@@ -1,35 +1,36 @@
 # frozen_string_literal: true
 
 class CinemaHallsController < ApplicationController
+  include JSONAPI::Fetching
   before_action :authenticate_user!
 
   def index
     cinema_halls = CinemaHall.all
 
-    render json: cinema_halls
+    render jsonapi: cinema_halls
   end
 
   def show
     cinema_hall = CinemaHall.find(params[:id])
 
-    render json: cinema_hall.to_json(only: %w[name row_number row_total_seats])
+    render jsonapi: cinema_hall, only: whitelisted_attributes
```

Comment:
After consideration, I realized that the better idea might be to explicitly tell what kind of attributes we want to blacklist instead of what kind of attributes might be whitelisted. It's of course just an idea but consider that you want to remove `created_at` and `updated_at` data from records. You will want to probably remove those fields also from other models. In that case, it might be easier to share this code between controllers instead of rewriting attributes from each one serializer in each one controller. 

```ruby
def index/show
   ...
   render jsonapi: collection, except: blacklisted_attributes
end

private

def blacklisted_attributes
  %w[created_at updated_at]
end
```

Remember it's just an idea which has also disadvantages, especially that it's better to tell explicitly what kind of data might be returned (explicit is always better than implicit communication), but from another side, you will have to always add everywhere additional method with listed attributes. And what's worst you will have to remember that adding new field it's not only adding it to `schema.rb`, but also to `serializer` and `controller`.

Well - as always it's not so clear and there is no good way. Just wanted to consider pros and cons of both solutions. Sorry for confusion 😅